src/index.js

```diff
@@ -120,11 +119,11 @@ export class SpectrumGenerator {
         : peakWidth,
       widthLeft,
       widthRight,
-      shape: shapeOptions,
+      shape: shapeOptions = {},
     } = options;
 
     if (peakOptions) {
-      Object.assign(shapeOptions || {}, peakOptions || {});
+      shapeOptions = Object.assign({}, shapeOptions, peakOptions);
```

Comment:
Please use the new syntax
```suggestion
      shapeOptions = {...shapeOptions, ...peakOptions};
```
`{...shapeOptions, ...peakOptions}`